components/layout/display_list_builder.rs

```diff
@@ -2963,6 +2971,27 @@ impl BlockFlowDisplayListBuilding for BlockFlow {
         state.processing_scrolling_overflow_element = false;
     }
 
+    fn block_stacking_context_type(
+        &self,
+        flags: StackingContextCollectionFlags,
+    ) -> BlockStackingContextType {
+        if flags.contains(NEVER_CREATES_STACKING_CONTEXT) {
+            return BlockStackingContextType::NonstackingContext;
+        }
+
+
+        if self.fragment.establishes_stacking_context() {
+            return BlockStackingContextType::StackingContext
+        }
+
+        if self.base.flags.contains(IS_ABSOLUTELY_POSITIONED) ||
+                self.fragment.style.get_box().position != position::T::static_ ||
+                self.base.flags.is_float() {
+            BlockStackingContextType::PseudoStackingContext
```

Comment:
Maybe this reads nicer returning early in each of those cases? Your call.

Reply:
Okay. This seems like an improvement to me as well.